src/api/spec/controllers/public_controller_spec.rb

```diff
@@ -1,14 +1,130 @@
 require 'rails_helper'
+# WARNING: If you change tests make sure you uncomment this line
+# and start a test backend. Some of the actions
+# require real backend answers for projects/packages.
+# CONFIG['global_write_through'] = true
 
 RSpec.describe PublicController, vcr: true do
-  let(:project) { create(:project, name: "public_controller_project") }
+  let(:project) { create(:project, name: "public_controller_project", title: "The Public Controller Project") }
   let(:package) { create(:package_with_file, name: "public_controller_package", project: project) }
 
   describe "GET #source_file" do
-    it "sends the requested file" do
+    before do
       get :source_file, project: project.name, package: package.name, filename: "somefile.txt"
-      expect(response).to have_http_status(:success)
-      expect(response.body).to eq(package.source_file("somefile.txt"))
     end
+
+    it { is_expected.to respond_with(:success) }
+    it { expect(response.body).to eq(package.source_file("somefile.txt")) }
+  end
+
+  describe "GET #index" do
+    before do
+      get :index
+    end
+
+    it { is_expected.to respond_with(302) }
```

Comment:
I'm not sure how rails handle this, but I agree if rails automatically sets the response code to `302` when redirecting a user to a page. But as far as I know, you can redirect a user even if the response code is `200` for example, but as I said, if rails automatically handles this, I agree.